app/src/index.jsx

```diff
@@ -10,14 +11,24 @@ import 'reset-css/reset.css';
 import 'common/css/fonts/fonts.scss';
 import 'common/css/common.scss';
 
+
 import App from './app';
+import store from './store';
 
 
 const queryParseHistory = qhistory(
   createHashHistory({ hashType: 'noslash' }),
   stringify,
   parse,
 );
+queryParseHistory.listen((location) => {
+  const match = matchPath(location.pathname, '/:projectId');
+  const hashProject = match.params.projectId;
+  const userProjects = Object.keys(userInfoSelector(store.getState()).assigned_projects);
+  if (userProjects.indexOf(hashProject) !== -1) {
```

Comment:
Please also do not dispatch the action if active project hasn't changed.

Reply:
fixed